src/listener_address.rs

```diff
@@ -52,6 +53,16 @@ pub enum ListenerAddress {
     ///
     /// Special name `*` means to bind all passed addresses simultaneously, if `multi-listener` crate feature is enabled.
     FromFdNamed(String),
+    /// The VSOCK address family facilitates communication between virtual machines and the host they are running on.
+    /// A socket address is defined as a combination of a 32-bit Context Identifier (CID) and a 32-bit port number.  The CID identifies
+    /// the source or destination, which is either a virtual machine or the host.
+    ///
+    /// [vsock manual]: https://www.man7.org/linux/man-pages/man7/vsock.7.html
+    /// (Implemented only on Linux and Darwin)
+    Vsock {
+        cid: u32,
```

Comment:
Documentation detailing what specifically `cid` and `port` mean can be moved here. Otherwise it triggers the `#[warn(missing_docs)]` lint.